src/component/ShoppingCart.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from "react";
+import Grid from "@mui/material/Grid";
+import ProductCard from "./ProductCard";
+import { Box } from "@mui/material";
+import PaginationComponent from "./Pagination.jsx";
+import { useShopping } from "../context/ShoppingContext.js";
+
+const ShoppingCart = ({ products }) => {
+  const { setOrder, order } = useShopping();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [product, setProduct] = useState(products);
+  const itemsPerPage = 12;
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+  let currentData = product.slice(startIndex, endIndex);
+  useEffect(() => {
+    currentData = product.slice(startIndex, endIndex);
+  }, [product]);
+  const handlePageChange = (newPage) => {
+    setCurrentPage(newPage);
+  };
+  const handleQuantityInc = (index) => {
+    setProduct(
+      product.map((prod) =>
+        prod.id === index
+          ? { ...prod, quantity: prod.quantity == 0 ? 1 : prod.quantity + 1 }
+          : { ...prod }
+      )
+    );
+  };
+  const handleQuantityDec = (index) => {
+    setProduct(
+      product.map((prod) =>
+        prod.id === index
+          ? { ...prod, quantity: prod.quantity == 0 ? 0 : prod.quantity - 1 }
+          : { ...prod }
+      )
+    );
+  };
+  const handleAddToCart = (index) => {
```

Comment:
can you find a more optimized data structure to store the products in a cart?

Reply:
Of course, in this case we are dealing with a product that is impossible to dauplicat. It is possible to use the map to increase performance, key maybe Uniqe as Name or ID from production